docs/design/README.md

```diff
@@ -256,6 +255,36 @@ etc.
 These metadata files will give twoliter the information it needs to resolve, pull, and aggregate all dependency kits and the correct SDK needed for a build.
 This dependency resolution will be modeled in Cargo and kept in a `Cargo.lock` file (see [Twoliter Update](#twoliter-update)).
 
+### Kit Aggregation when Building Variants
+
+A user may depend on a variable number of kits (i.e. one build may require 2 kits, another 3 kits, etc.).
+The variant build process relies on a Dockerfile, which cannot accept or copy RPMs from a variable number of container images.
+Thus, before a variant image can be built, we will need to aggregate the kits that it depends on.
+
+Once local kits are built and external kits are pulled, we will copy each kit into a single container image.
+We will do this by generating a `Dockerfile` that looks like this:
+
+```Dockerfile
+FROM scratch
+
+# External Kits pulled from registries
+# TODO - is the registry name part of the path or kit name?
+COPY --from=kit1-x86_64:v0.1.0 /kit-data /kits/x86_64/kit1-v0.1.0
+COPY --from=kit2-x86_64:v0.2.0 /kit-data /kits/x86_64/kit2-v0.2.0
+
+# Local kits built on the host
+COPY ./build/kits/x86_64/mylocalkit1-v0.1.0 /kits/x86_64/mylocalkit1-v0.1.0
+COPY ./build/kits/x86_64/mylocalkit2-v0.2.0 /kits/x86_64/mylocalkit1-v0.2.0
```

Comment:
Summarizing discussion - the monolithic kit-of-kits approach will be challenging with local kits that depend on other local kits, where we need to instantiate the image multiple times.